web/src/test/java/org/cbioportal/web/GenesetDataControllerTest.java

```diff
@@ -64,46 +64,77 @@ public void setUp() throws Exception {
-                .andExpect(MockMvcResultMatchers.status().isOk())
-                .andExpect(MockMvcResultMatchers.content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
-                .andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(2)))
-                .andExpect(MockMvcResultMatchers.jsonPath("$[0].geneticProfileId").value(PROF_ID))
-                .andExpect(MockMvcResultMatchers.jsonPath("$[0].genesetId").value(GENESET_ID_1))
-                .andExpect(MockMvcResultMatchers.jsonPath("$[0].sampleId").value(SAMPLE_STABLE_ID_1))
-                .andExpect(MockMvcResultMatchers.jsonPath("$[0].value").value(VALUE_1))
-                .andExpect(MockMvcResultMatchers.jsonPath("$[1].geneticProfileId").value(PROF_ID))
-                .andExpect(MockMvcResultMatchers.jsonPath("$[1].genesetId").value(GENESET_ID_2))
-                .andExpect(MockMvcResultMatchers.jsonPath("$[1].sampleId").value(SAMPLE_STABLE_ID_1))
-                .andExpect(MockMvcResultMatchers.jsonPath("$[1].value").value(VALUE_2));
-
-        //testing the sampleListId route:
-        // List<GenesetMolecularData> genesetDataItems2 = createGenesetDataItemsList();
-        // genesetDataItems2.addAll(createGenesetDataItemsList());//duplicate, just to make it different from response above
-        // Mockito.when(genesetDataService.fetchGenesetData(Mockito.any(), (List) Mockito.any(),
-        //     Mockito.any())).thenReturn(genesetDataItems2);
-        // //set sampleListId to ensure the fetchGenesetData variant above is called:
-        // genesetDataFilterCriteria.setSampleListId(SAMPLE_LIST_ID);
-
-        // mockMvc.perform(MockMvcRequestBuilders.post("/genetic-profiles/" + PROF_ID + "/geneset-genetic-data/fetch")
-        //         .accept(MediaType.APPLICATION_JSON)
-        //         .contentType(MediaType.APPLICATION_JSON)
-        //         .content(objectMapper.writeValueAsString(genesetDataFilterCriteria)))
-        //         .andExpect(MockMvcResultMatchers.status().isOk())
-        //         .andExpect(MockMvcResultMatchers.content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
-        //         .andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(4)));
+        genesetDataFilterCriteria.setSampleListId(sampleListId);
+        genesetDataFilterCriteria.setGenesetIds(genesetIds);
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/genetic-profiles/" + geneticProfileId + "/geneset-genetic-data/fetch")
+            .accept(MediaType.APPLICATION_JSON)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(objectMapper.writeValueAsString(genesetDataFilterCriteria)))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andExpect(MockMvcResultMatchers.content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(2)))
```

Comment:
Why did this change from 4 to 2? I liked that the two tests verified different results. It helped mentally separate the two tests for me